### PR TITLE
Remove public_assets.php — serve directly from filesystem

### LINKED ISSUE #144: public_assets.php assumes local file storage, incompatible with S3
## Problem

`public_assets.php` serves files directly from the local filesystem, assuming that public assets are stored in a local directory. When using S3 or Swift file backends via the AWS extension, assets are not on the local filesystem and this path fails.

This is the same class of issue as canasta_img.php (#142) — direct filesystem access instead of using MediaWiki's file backend API.

## Proposed Fix

Use MediaWiki's file backend API (`RepoGroup` or equivalent) to resolve and serve files, respecting the configured storage backend (local, S3, Swift).

### PR DESCRIPTION
**Pull request by cicalese**
## Summary

`public_assets.php` read `wikis.yaml` on every request to determine the wiki ID, then streamed files from `/mediawiki/public_assets/<wikiId>/`. This hardcoded a local-filesystem assumption that broke any deployment where public assets live in S3 / Swift / a network mount.

The wiki-ID-from-URL lookup that the script did per request can be done **once at container startup** from the same `wikis.yaml`. Apache then serves the files natively — no PHP round trip per asset, no custom code path that needs to be S3-aware.

Fixes #144. Same shape as #147 (canasta_img.php removal), one layer down in the stack.

## What changed

- **Deleted**: `_sources/canasta/public_assets.php` (133 lines)
- **`_sources/configs/.htaccess`**: removed the `^/?w/public_assets.php/` passthrough rewrite rule
- **`Dockerfile`**:
  - Removed the `COPY` of `public_assets.php`
  - Removed the two static `Alias /public_assets/...` lines that pointed at the PHP entry point
  - Added a build-time `<Directory /mediawiki/public_assets>` block granting read access to the path tree
- **`_sources/scripts/config-subdir-wikis.sh`**: rewritten to walk `wikis.yaml` properly (id+url pairs via awk), and for each wiki emit a per-wiki `RewriteCond`/`RewriteRule` keyed off `%{HTTP_HOST}` that maps the wiki's `/public_assets/...` URL to the matching filesystem directory at `/mediawiki/public_assets/<wiki_id>/`. Subdir-wiki plumbing (symlinks, `.htaccess` rewrites, `img_auth.php` aliases) is preserved.

The generated Apache config for a two-wiki setup looks like:

```apache
<Directory /mediawiki/public_assets>
    Require all granted
    Options -Indexes
</Directory>
RewriteCond %{HTTP_HOST} ^localhost:8080$ [NC]
RewriteRule ^/public_assets/(.*)$ /mediawiki/public_assets/main/$1 [L]
RewriteCond %{HTTP_HOST} ^localhost:8080$ [NC]
RewriteRule ^/second/public_assets/(.*)$ /mediawiki/public_assets/second/$1 [L]
```

The `HTTP_HOST` condition makes this safe for multi-host wiki farms — only the matching host's request paths get routed to that wiki's storage directory. Single-host setups (the common case) work the same way; the condition is just always satisfied.

## Why this resolves #144

Once Apache is doing the serving, "store public assets in S3" becomes:

1. Mount the bucket somewhere on the filesystem (`s3fs`, `goofys`, `mountpoint-s3`, a CSI driver, etc.)
2. The existing rewrite target uses it as-is

No Canasta-specific storage adapter required. Same shape the AWS extension takes for wiki uploads, just one layer lower in the stack — Apache instead of MediaWiki.

Note: this issue does not mention the AWS extension, and the AWS extension wouldn't have been the right tool for `public_assets` anyway (it's a `FileRepo` adapter, and `public_assets` isn't a `FileRepo`). The unified-S3 story is "AWS extension handles wiki uploads, filesystem mount handles `public_assets`, both can live in the same bucket under different prefixes."

## Verified end-to-end

Built a fresh canasta image via `--build-from` against a two-wiki test instance (main at root, second at `/second`), placed test files in each wiki's `public_assets/` directory, and ran the full matrix:

| Endpoint | Wiki | Expected | Result |
|---|---|---|---|
| `/public_assets/logo.png` | main | 200 + main content | ✓ |
| `/second/public_assets/logo.png` | second | 200 + second content | ✓ |
| `/public_assets/missing.png` | main | 404 | ✓ |
| `/public_assets/` (directory listing) | — | 403 (`Options -Indexes`) | ✓ |
| URL-encoded `..` traversal | — | 404 (Apache normalizes before mod_rewrite) | ✓ |
| `/etc/passwd` traversal attempt | — | 404 | ✓ |
| `/w/images/...` anon (regression check from #147) | main | 403 (img_auth.php) | ✓ |
| `/second/w/images/...` anon | second | 403 | ✓ |

`img_auth.php` behavior from #147 is preserved — anonymous image access on private wikis still returns 403. The `<Directory /mediawiki/public_assets>` grant is scoped to that path tree only and does not affect the wiki's `images/` directory (which lives under a different mount and remains gated by `img_auth.php`).

## Test infrastructure

This PR also adds the first test suite to CanastaBase since the unused PHPUnit scaffolding was removed in #147:

- **`tests/test_config_subdir_wikis.py`** — 16 pytest tests that run the real `config-subdir-wikis.sh` in a temp workspace via subprocess and assert on the resulting Apache config and filesystem state. Coverage:
  - **Apache rewrite generation (7 tests):** root wikis, subdir wikis, dot-escaping, ports in `HTTP_HOST`, port+subdir combos, multi-host farms, exactly-one-rewrite-per-wiki
  - **Subdir-wiki plumbing (6 tests):** symlink creation, `.htaccess` rewriting, `img_auth.php` Aliases, no-op for root wikis, dedup of repeated subdirs across hosts
  - **Edge cases (3 tests):** empty `wikis.yaml`, existing `apache2.conf` content preserved, wiki ids with underscores
- **`tests/conftest.py`** — `workspace` fixture (mirrors the canasta-image layout in a temp dir) and `script_runner` helper
- **`requirements-test.txt`** — just `pytest>=8.0`
- **`.github/workflows/tests.yml`** — runs the suite on push/PR to master

Two small refactors to the script for testability (production behavior unchanged):

1. `WIKIS_YAML` and `APACHE_CONF` are now env-var driven, defaulting to the canonical canasta locations.
2. Replaced `declare -A` (bash 4) with a delimited string for bash 3.x compatibility — macOS still ships bash 3.2, and requiring bash 4 to run tests locally would be friction.

PHPUnit can layer on top of this later via #130 when there's PHP code that needs unit testing.

## Test plan

- [x] Wiki at root: public_assets serves
- [x] Wiki at subdir: public_assets serves with the right wiki's content
- [x] Multi-host config syntax (`HTTP_HOST` condition) generated correctly per wiki
- [x] Missing files return 404 (not the wiki's catch-all)
- [x] Directory listing disabled
- [x] Path traversal (literal `..`, URL-encoded `..`, encoded slash) blocked
- [x] `img_auth.php` behavior unchanged (regression check on #147's matrix)
- [x] 16 unit tests passing locally and in CI
- [x] CI passes (Unit Tests, Docker build, Docker push)
- [ ] Verified with an actual S3 mount (s3fs / mountpoint-s3) — out of scope here, but the path is now "mount the bucket and re-create the instance"

## Net change

```
 .github/workflows/tests.yml             |  25 ++++
 .gitignore                              |   6 +
 Dockerfile                              |   9 +-
 _sources/canasta/public_assets.php      | 133 ---------- (deleted)
 _sources/configs/.htaccess              |   3 -
 _sources/scripts/config-subdir-wikis.sh | 127 +++++++++++++++---
 requirements-test.txt                   |   1 +
 tests/conftest.py                       | 109 +++++++++++++
 tests/test_config_subdir_wikis.py       | 186 ++++++++++++++++++++++
 9 files changed, 428 insertions(+), 182 deletions(-)
```

Production code: **−75 net lines** (133 deleted, 58 net added across Dockerfile/script/.htaccess), plus the operational gains: faster requests (no PHP per asset), native Apache cache headers / `mod_expires` / range-request handling, and a storage backend that's now a config knob instead of a Canasta concern.

Test infrastructure: +330 lines (one-time investment, covers the script comfortably and lays groundwork for #130).

### CODE REVIEW

File: _sources/scripts/config-subdir-wikis.sh [hunk 1]
```diff
@@ -1,37 +1,99 @@
...
+        ln -sf "$MW_HOME" "$WWW_ROOT/$wiki_path"
+
+        sed -e "s|w/rest.php/|$wiki_path/w/rest.php/|g" \
+            -e "s|w/img_auth.php/|$wiki_path/w/img_auth.php/|g" \
+            -e "s|^/*\$ %{DOCUMENT_ROOT}/w/index.php|/*\$ %{DOCUMENT_ROOT}/$wiki_path/w/index.php|" \
+            -e "s|^\\(.*\\)\$ %{DOCUMENT_ROOT}/w/index.php|\\1\$ %{DOCUMENT_ROOT}/$wiki_path/w/index.php|" \
+            "$WWW_ROOT/.htaccess" > "$WWW_ROOT/$wiki_path/.htaccess"
+
+        echo "Alias /$wiki_path/w/images/ /var/www/mediawiki/w/img_auth.php/" >> "$APACHE_CONF"
+        echo "Alias /$wiki_path/w/images /var/www/mediawiki/w/img_auth.php" >> "$APACHE_CONF"
```
Comment (yaronkoren) [inline _sources/scripts/config-subdir-wikis.sh:97]: Is this second line necessary?

**Review by yaronkoren — APPROVED** [review]
Looks awesome! Great additional tests as well.
